### PR TITLE
Tweak ConfigNotFoundException class

### PR DESCRIPTION
**Pull request by ChristopheDuong**
## What
The exception has constructors that require a first argument of `ConfigSchema`:
`ConfigNotFoundException(ConfigSchema type, UUID uuid)` 

## How
Make it accept more than just `ConfigSchema` in case we are loading other configuration objects.

### CODE REVIEW

File: airbyte-config/models/src/main/java/io/airbyte/config/Configs.java [hunk 1]
```diff
@@ -31,6 +31,8 @@
 
   String getAirbyteRole();
 
+  String getAirbyteServerHost();
```
Comment (subodh1810) [inline airbyte-config/models/src/main/java/io/airbyte/config/Configs.java:34]: Why this change?

Comment (ChristopheDuong) [reply airbyte-config/models/src/main/java/io/airbyte/config/Configs.java:34]: I need it in the cloud to know where the airbyte server is deployed to call the API

**Review by subodh1810 — APPROVED** [review]

**Review by davinchia — APPROVED** [review]
One change around using `INTERNAL_API_HOST` instead of creating a new env var.
